packages/grafana-ui/src/components/uPlot/Plot.tsx

```diff
@@ -40,6 +49,7 @@ export const UPlotChart: React.FC<PlotProps> = props => {
     // 1. When config is ready and there is no uPlot instance, create new uPlot and return
     if (isConfigReady && !plotInstance.current) {
       plotInstance.current = initializePlot(prepareData(props.data), currentConfig.current, canvasRef.current);
+      updateTimeScale();
```

Comment:
This feels messy to have to set this directly of init, can't we initialize the time scale with the correct (initial) min/max values when we defined the scale?
https://github.com/grafana/grafana/blob/master/packages/grafana-ui/src/components/GraphNG/GraphNG.tsx#L80

Reply:
I think the problem will be the invalidation of the config in such case, which is sub optimal as it will reinitialize the entire plot. 

But ref https://github.com/leeoniya/uPlot/blob/005b3e0903444521f592c14cf33ab922d6e364ca/dist/uPlot.d.ts#L156 range of a scale can be defined as a function, then we could extend the UPlotConfigBuilder.addScale interface to accept the range (if it isn't accepting it already), and using references retrieved it in the memoized builder. 

A draft of this approach would be something like this:

```ts
const GraphNG = () => {
  // reference change will not triger re-render
  const currentTimeRange = useRef();
  // keep reference value in sync with the props
  useEffect(() => {
    currentTimeRange.current = props.timeRange;
  }, [props.timeRange]);

  const configBuilder = useMemo(() => {
    const builder = new UPlotConfigBuilder();

    builder.addScale({
      // lazily access reference value without adding dependency to props.timeRange to the memoization
      range: () => [timeRange.current.from.valueOf(), timeRange.current.to.valueOf()],
    });
    return builder;
  }, [configRev, timeZone]);
}
```

Reply:
@dprokop :100: what i was trying to do.

Reply:
> I think the problem will be the invalidation of the config in such case 

But this case is for when the plot is re-initialized, which means the config is new so not sure that is a problem

Reply:
defining `range` as a function makes this distinction i irrelevant, as long as it's always able to return the correct current values.